src/lib/domutil.js

```diff
@@ -2,49 +2,82 @@ import {getElement, isString, isEmptyString, isFunction} from '@/lib/helper';
 
 let addEvent = null;
 let removeEvent = null;
-let element = null; // TODO: 초기화
-let listeners = []; // TODO: 초기화, 요소-타입별 관리
+let element = null;
+
+const listenersHelper = {
+  listeners: new Map(),
+  add(element, type, handler) {
+    const handlers = listenersHelper.getHandlers(element, type);
+    handlers.push(handler);
+  },
+  remove(element, type, handler) {
+    const handlers = listenersHelper.getHandlers(element, type);
+    listenersHelper.setHandlers(
+      element,
+      type,
+      handlers.filter((f) => f !== handler)
+    );
+  },
+  getEventTypes(element) {
+    // return : Map { click => [], focus => [], ... }
+    if (listenersHelper.listeners.has(element)) return listenersHelper.listeners.get(element);
+    else return listenersHelper.listeners.set(element, new Map()).get(element);
```

Comment:
지극히 개인적인 생각으로는 bracket을 추가하는게 가독성이 더 좋은 것 같아요!

```
    if (listenersHelper.listeners.has(element)) {
      return listenersHelper.listeners.get(element);
    } else {
      return listenersHelper.listeners.set(element, new Map()).get(element);
    }
```

또는 다음과같이 작성해도 좋을 것 같습니다!

```suggestion
    if (!listenersHelper.listeners.has(element)) {
      return listenersHelper.listeners.set(element, new Map()).get(element);
    } 
    
    return listenersHelper.listeners.get(element);
```

Reply:
- 약간 개인 차이가 존재하는 부분일거 같은데, 비슷한 경험이 있었던 기억이 나서 코멘트 남겨요.
  - 저도 if else 체크해야 할 경우, 처리할게 단순한 것을 if 체크 후 return 으로 조건을 끊어내는 방식을 좋아합니다.
  - 이렇게 작성할 경우 if / else 로 인한 들여쓰기가 조금 더 줄어들어 개인적으로 가독성이 좋다고 생각합니다.
  - 이런 처리를 했을때 받았던 피드백은 return 으로 조건을 끊어 내면, 코드를 바로 이해하기 어렵다는 피드백도 받은 기억이 있습니다.
    - 이 피드백도 납득이 되는 부분으로 기억합니다.
    - 피드백을 받고 보완해봤던 방식은, 변수명을 명시적으로 선언해보는 방식도 써보고 있습니다.
```javascript
const isEmptyEventTypes = !listenersHelper.listeners.has(element);
if (isEmptyEventTypes) {
 return listenersHelper.listeners.set(element, new Map()).get(element);
}

return listenersHelper.listeners.get(element);
```

Reply:
제가 가볍게 넘어갔던 부분을 지적해주셔서 감사합니다.
예전에는 준우님 방식으로 return이 같은 들여쓰기 라인에 있도록 정렬하는게 좋아서 그렇게 했었는데요. 
현관님이 제안해주신 방식이 제가 어떤 내용의 조건을 판단하는 건지 더 직관적인거 같네요.